### ISSUE TITLE
Feature/request Add formatting and field separator flags/options.

### ISSUE THREAD

**Issue by raiyuza** [tmuxinator/tmuxinator#681]
**Is your feature request related to a problem? Please describe.**
We have too little control over the output generated from the list command. This is referenced in issue [678](https://github.com/tmuxinator/tmuxinator/issues/678) and [this](https://github.com/tmuxinator/tmuxinator/pull/679) pull request. 

Quite a lot of parser and lexers, depend on a preformatted data structure, we need to do a lot of shell magic to get the output we want.

For Ex.
```
mux list | awk -F "\t" "{ print $1 }"
```

You would expect the output to be listed like ``ls -1``, but since it doesn't recognise the default field separator as tab. It just show the default listing behavior.

**Describe the solution you'd like**
We can keep the default behavior by giving it a seperator flag ala awk -F though. Users can set an env variable to control the default value of this flag. This way you don't have to break backwards compatibility and it makes the feature more useful overall.

In the same fashion it would be amazing to have have some kind of --format option, it would be able to
format how the list command would show your list, and extract properties from said projects. Ex.

```
$> mux l -F "\n" --format "@name\t@root\t@windows.editor.layout"
$> 
foo<tab>~/foo<tab>vertical
bar<tab>~/bar<tab>horizontal
```

This opens up an powerful way to create a formatted data structure. This mainly used for parsers in Ex.

* vim, you could write a unite/denite script to read the outputs buffer to make menu's
* fuzzy search

tl;dr
1. Add -F flag that somewhat works like awk -F
2. Add a --format flag that defines the value/line returned as defined by it's param(string/syntax!?)

**Describe alternatives you've considered**
1. Making my own PRQ
2. Accept behavior and use a lot of shell magic to get the output formatted as we want.

**Additional context**
I am slightly conflicted here, as the linux mantra says. 

> Write programs that do one thing and do it well.
> Write programs to work together.
> Write programs to handle text streams, because that is a universal interface

We _should_ be able to use awk and sed to extract patterns already from the default list format, this is not functioning as we/I want it. 

Though, number 2 also tells us to make programs work together. It's working pretty crappy together right now IMHO. 

Thoughts?

PS. A option to remove the ``tmuxinator projects:`` string as a flag would be nice too. ``ls`` doesn't print any extra text, saves us a head/tail pipe again!

**Comment by raiyuza** [comment]
Any feedback would be nice so i can start implementing this somewhere this week !

**Comment by ethagnawl** [comment]
My concern here is that this feature adds complexity in order to cover a use case which is -- in all likelihood -- very uncommon.

Perhaps a better, catch-all solution (as discussed in #679) would be to remove the `tmuxinator projects:` header and change the default behavior to return the list of projects separated by newlines?

**Comment by ethagnawl** [comment]
Does anyone else have thoughts on the above? (CC @tmuxinator/tmuxinator)

**Comment by raiyuza** [comment]
@ethagnawl , Apparently my functionality got broken. Ill fix it somewhere this week!
